core/src/main/java/org/apache/carbondata/core/memory/UnsafeMemoryManager.java

```diff
@@ -173,6 +174,7 @@ public synchronized void freeMemoryAll(String taskId) {
           "Freeing offheap working memory of size %d. Current available memory is %d",
           occuppiedMemory, totalMemory - memoryUsed));
     }
+    ThreadLocalTaskInfo.clearCarbonTaskInfo();
```

Comment:
Better call explicitly from methods where we call `freeMemoryAll, thread clearing should not be part of it.

Reply:
OK..